src/bmi.jl

```diff
@@ -65,10 +65,20 @@ end
 
 function BMI.update_until(model::Model, time::Float64)
     @unpack clock, network, config = model
-    curtime = BMI.get_current_time(model)
-    n = Int(max(0, (time - curtime) / model.clock.Δt.value))
-    for _ = 1:n
-        model = run_timestep(model)
+    t = BMI.get_current_time(model)
+    steps = (time - t) / model.clock.Δt.value
+    if steps < 0
+        error("The current model timestamp $t is larger than provided `time` $time")
+    elseif !isinteger(steps)
+        error_message = string(
+            "Provided `time` $time minus the current model timestamp $t",
+            " is not an integer multiple of model time step $(model.clock.Δt.value)",
+        )
+        error(error_message)
+    else
+        for _ = 1:steps
+            model = run_timestep(model)
+        end
```

Comment:
`else` is never necessary.
`else` is preferable if there is high cohesion among the branches.
In this case the first two branches throw errors and the final branch advances the model in time.
Due to low cohesion I suggest to take the loop out of the conditional block.

Reply:
Good point, see commit https://github.com/Deltares/Wflow.jl/pull/387/commits/da8948cd5dca574e049d3b43b1feace77dda4a16.